packages/block-editor/src/components/block-list/block.js

```diff
@@ -241,30 +207,28 @@ function BlockListBlock( {
 	} else if ( blockType?.apiVersion > 1 ) {
 		block = blockEdit;
 	} else {
-		block = <Block { ...wrapperProps }>{ blockEdit }</Block>;
+		block = <Block>{ blockEdit }</Block>;
 	}
 
 	const { 'data-align': dataAlign, ...restWrapperProps } = wrapperProps ?? {};
 
-	const value = {
-		clientId,
-		className: classnames(
-			{
-				'is-editing-disabled': blockEditingMode === 'disabled',
-				'is-content-locked-temporarily-editing-as-blocks':
-					isTemporarilyEditingAsBlocks,
-			},
-			dataAlign && themeSupportsLayout && `align${ dataAlign }`,
-			! ( dataAlign && isSticky ) && className
-		),
-		wrapperProps: restWrapperProps,
-		isAligned,
-	};
-
-	const memoizedValue = useMemo( () => value, Object.values( value ) );
+	restWrapperProps.className = classnames(
+		restWrapperProps.className,
+		dataAlign && themeSupportsLayout && `align${ dataAlign }`,
+		! ( dataAlign && isSticky ) && className
+	);
 
+	// We set a new context with the adjusted and filtered wrapperProps (through
+	// `editor.BlockListBlock`), which the `BlockListBlockProvider` did not have
+	// access to.
 	return (
-		<BlockListBlockContext.Provider value={ memoizedValue }>
+		<PrivateBlockContext.Provider
+			value={ {
+				wrapperProps: restWrapperProps,
+				isAligned,
+				...context,
+			} }
```

Comment:
@ellatrix, can we add a comment here explaining why there's no need to memoize the `BlockContext` value?